ml-prims/src/linalg/svd.h

```diff
@@ -88,32 +90,28 @@ void svdQR(T *in, int n_rows, int n_cols, T *sing_vals, T *left_sing_vecs,
   CUDA_CHECK(cudaGetLastError());
 
   int dev_info;
-  updateHost(&dev_info, devInfo, 1);
+  updateHostAsync(&dev_info, devInfo.data(), 1, stream);
+  CUDA_CHECK(cudaStreamSynchronize(stream));
   ASSERT(dev_info == 0,
          "svd.h: svd couldn't converge to a solution. "
          "This usually occurs when some of the features do not vary enough.");
-  
-  ///@todo: what if stream from cusolver handle is different!?
-  mgr.free(devInfo, stream);
-  mgr.free(d_work, stream);
 }
 
 template <typename T>
 void svdEig(T* in, int n_rows, int n_cols, T* S,
-		   T* U, T* V, bool gen_left_vec,
+            T* U, T* V, bool gen_left_vec,
             cublasHandle_t cublasH, cusolverDnHandle_t cusolverH,
-            cudaStream_t stream, DeviceAllocator &mgr) {
+            cudaStream_t stream, std::shared_ptr<deviceAllocator> allocator) {
```

Comment:
Could pass in cumlHandle_impl instead of cuBLAS handle, cuSolver handle and allocator.

Reply:
will be discussed in #498.